components/dashboard/package.json

```diff
@@ -71,7 +71,7 @@
   "scripts": {
     "start": "craco start",
     "build": "craco build --verbose",
-    "lint": "eslint --ext=.jsx,.js,.tsx,.ts ./src",
+    "lint": "eslint --max-warnings=0 --ext=.jsx,.js,.tsx,.ts ./src",
```

Comment:
Can we change it such that it applies the linter to only modified files? I believe that should be done by passing the files into this command, and in the pre-commit enabling the `pass_filenames` feature. 

That way, we don't need to restrict it to just dashboard, but can enable for all ts/js files in incremental mode

Reply:
Discussed in [this loom](https://gitpod.slack.com/archives/C02EN94AEPL/p1682376030511889), but sounds like it's not straight forward.

But would be great to have!

Reply:
Maybe we already tried, but [this](https://gist.github.com/seeliang/0f0de424d1cdc4541c338f4ee93b7e6a#ci-script) looks promising.

Reply:
Let's land as is, and we can do a follow-up.

Reply:
I've tweaked things in the leeway command to use git diff to help filter out the files to run against and it seems to work as we want.

https://github.com/gitpod-io/gitpod/pull/17355/files#diff-b5cfb13a17f4f90cdd041c23e0b5ff56941b32cd8def8a743476e0cc0944b5cbR56